### PR TITLE
Add series to the search results.

### PR DESCRIPTION
**Pull request by kadams54**
Fixes https://github.com/juju/juju-gui/issues/1587.

### CODE REVIEW

**Comment by kadams54** [comment]
## QA

1. Search for something.
2. Verify that the series are listed for all charms (but not bundles).
3. Verify that clicking on the series takes you to the charm details for that specific charm (i.e., click on "trusty" does not take you to the "precise" version of the charm).
4. Validate all of the above for both recommended and community entities.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4912/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4913/
Test PASSed.

**Comment by bac** [comment]
QA:

Searched for 'charmworld'. Selected to view charms only.  On Rabbitmq clicked on Precise.  Got http://salchicha.local:6543/?store=trusty/rabbitmq-server

Also I see the series designations are now all black, the orange is gone.

File: jujugui/static/gui/src/app/components/search-results-item/search-results-item.js [hunk 1]
```diff
@@ -93,6 +93,33 @@ YUI.add('search-results-item', function(Y) {
...
+      var components = [];
+      // Prevent layouts from collapsing due to empty content.
+      if (series.length === 0) {
+        return <span>{' '}</span>;
+      }
+      series.forEach(function(s) {
+        components.push(
+          <li className="list-series__item"
+            key={s.name}>
+            <a onClick={this._handleItemClick.bind(this, item.id)}>
```
Comment (bac) [inline jujugui/static/gui/src/app/components/search-results-item/search-results-item.js]: Need some series goo in here, I suspect.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4932/
Test PASSed.

**Comment by Lucapaulina** [comment]
We did a design for this, you can see it here: https://raw.githubusercontent.com/CanonicalLtd/juju-design/master/Search/Search_results.png?token=AFme8iCd5XMnOanPnVhcj1y_XKri9Eoeks5XLKchwA%3D%3D

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4934/
Test PASSed.

**Comment by kadams54** [comment]
Thanks @Lucapaulina - I'll update to match the design.

File: jujugui/static/gui/src/app/components/search-results-item/search-results-item.js [hunk 2]
```diff
@@ -93,6 +93,40 @@ YUI.add('search-results-item', function(Y) {
...
+      var series = item.series;
+      var components = [];
+      // Prevent layouts from collapsing due to empty content.
+      if (series.length === 0) {
+        return <li>&nbsp;</li>;
+      }
+      // This presumes that the series are sorted with most recent first (which
+      // they are) and that most recent is the default (which it is).
+      var defaultSeries = series[0].name;
+      series.forEach(function(s) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/search-results-item/search-results-item.js]: you can use an arrow fn here

File: jujugui/static/gui/src/app/components/search-results-item/search-results-item.js [hunk 3]
```diff
@@ -93,6 +93,40 @@ YUI.add('search-results-item', function(Y) {
...
+        if (defaultSeries !== s.name) {
+          id = id.replace(defaultSeries + '/', s.name + '/');
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/search-results-item/search-results-item.js]: I have no idea what's going on here or why this is necessary. Can you add a code comment

**Comment by bac** [comment]
QA notes

* Search for `ceph`. Find the one for -next that has many series including Xenial and Wily. Click on any series but Trusty.  All lead to `http://salchicha.local:6543/?store=~openstack-charmers-next/trusty/ceph`

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4964/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4965/
Test PASSed.

**Comment by kadams54** [comment]
Ready for QA round 2. I also updated to match the design from @Lucapaulina, though we still have issues with wrapping once there are more than three series. I played around a bit trying to solve that, but stealing width from other columns just created wrapping issues in other places.

File: jujugui/static/gui/src/app/components/search-results-item/search-results-item.js [hunk 4]
```diff
@@ -93,6 +93,33 @@ YUI.add('search-results-item', function(Y) {
...
+      var components = [];
+      // Prevent layouts from collapsing due to empty content.
+      if (series.length === 0) {
+        return <li>&nbsp;</li>;
+      }
+      series.forEach(function(s) {
+        components.push(
+          <li className="list-series__item"
+            key={s.name}>
+            <a onClick={this._handleItemClick.bind(this, s.storeId)}>
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/search-results-item/search-results-item.js]: Should this have an empty href to be valid HTML?

**Comment by anthonydillon** [comment]
Code looks good apart from one small question. :+1: 

Can I recommend we render a different list item based on the item type (charm or bundle). As bundles have many charms and no series it makes sense to give the charm list a `three-col` but for charms we only have one so the charm list might as well be `one-col` which means you can make the series column `four-col`.

That should give use enough room to show all series in one line for now.

**Comment by hatched** [comment]
Thanks everyone for working through this one. @anthonydillon I agree with your suggestion can you tackle that task as a follow-up to this when you have a chance? 

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**